### PR TITLE
Hemophages Metabolize Blood Faster (& Better)

### PR DESCRIPTION
**Pull request by Nerev4r**
## About The Pull Request
This makes hemophages metabolize blood faster in their tummies. Also, this makes them actually metabolize it from food. Who knew.

## How This Contributes To The Skyrat Roleplay Experience
Allows better ingestion of the liquid stuff.

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
 Over in the last PR, but also.
 
 
image

 
</details>

## Changelog
:cl:
balance: Hemophages now digest blood faster.
fix: Hemophages now actually digest blood that's in a food.
/:cl:

### CODE REVIEW

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm [hunk 1]
```diff
@@ -55,6 +55,8 @@
 /// The ratio of reagents that get purged while a Hemophage vomits from trying to eat/drink something that their tumor doesn't like.
 #define HEMOPHAGE_VOMIT_PURGE_RATIO 0.95
 
+/// The rate at which blood metabolizes in a Hemophage's stomach subtype.
+#define BLOOD_METABOLIZATION_RATE 0.1 * REAGENTS_METABOLISM
```
Comment (Zonespace27) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm]:
```suggestion
#define BLOOD_METABOLIZATION_RATE (0.1 * REAGENTS_METABOLISM)
```
good form and to make sure intended order of operations is followed

Comment (Nerev4r) [reply modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm]: thank you.

File: modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm [hunk 2]
```diff
@@ -472,6 +474,15 @@
...
 	return ..()
 
+/obj/item/organ/internal/stomach/hemophage/on_life(delta_time, times_fired)
+	var/datum/reagent/blood/blood = reagents.get_reagent(/datum/reagent/blood)
+	if(blood)
+		var/amount_max = blood.volume
+		var/amount = min((round(metabolism_efficiency * amount_max, 0.05) + BLOOD_METABOLIZATION_RATE) * delta_time, amount_max)
+
+		reagents.trans_id_to(owner, blood.type, amount=amount)
+	else ..()
```
Comment (tf-4) [inline modular_skyrat/modules/customization/modules/mob/living/carbon/human/species/hemophage.dm]: do you not want this to be `..()`? Or it'll skip processing all reagents if there's blood present

**Review by tf-4 — APPROVED** [review]

**Comment by savoryrex** [comment]
:)

**Pull request merged**